src/containers/CreateCommunity.js

```diff
@@ -0,0 +1,434 @@
+}))
+export class CreateCommunityInvite extends React.Component {
+
+  static propTypes = {
+    community: object,
+    currentUser: object,
+    invitationEditor: object,
+    dispatch: func
+  }
+
+  constructor (props) {
+    super(props)
+    this.state = {
+      emails: '',
+      expanded: false
+    }
+  }
+
+  processCSV () {
+    const { dispatch } = this.props
+
+    const appendComma = string => {
+      if (isEmpty(string) || string.trim().slice(-1) === ',') return string
+      return string + ','
+    }
+
+    emailsFromCSVFile(this.refs.fileInput.files[0])
+    .then(emails => {
+      const textarea = this.refs.emails
+      dispatch(updateInvitationEditor('recipients',
+        appendComma(textarea.getValue()) + ' ' + emails.join(', ')))
+    })
+  }
+
+  componentDidMount () {
+    const { dispatch, community, invitationEditor } = this.props
+    const { subject, message } = invitationEditor
+
+    if (subject === undefined) {
+      dispatch(updateInvitationEditor('subject', defaultSubject(community.name)))
+    }
+    if (message === undefined) {
+      dispatch(updateInvitationEditor('message', defaultMessage(community.name)))
+    }
+  }
+
+  render () {
+    const { expanded } = this.state
+    const { community, currentUser, dispatch, invitationEditor } = this.props
+
+    if (!canInvite(currentUser, community)) {
+      return <div>
+        You don&#39;t have permission to view this page.
+      </div>
+    }
+
+    let { subject, message, recipients, moderator, error } = invitationEditor
+
+    let setError = text => dispatch(updateInvitationEditor('error', text))
+
+    let update = (field, toggle) => event => {
+      let { value } = event.target
+      if (toggle) value = !invitationEditor[field]
+      dispatch(updateInvitationEditor(field, value))
+    }
+
+    let submit = () => {
+      dispatch(updateInvitationEditor('results', null))
+      setError(null)
+
+      if (!subject) return setError('The subject may not be blank.')
+      if (!message) return setError('The message may not be blank.')
+
+      let emails = parseEmailString(recipients)
+      if (isEmpty(emails)) return setError('Enter at least one email address.')
+
+      let badEmails = emails.filter(email => !validator.isEmail(email))
+      if (some(id => id, badEmails)) return setError(`These emails are invalid: ${badEmails.join(', ')}`)
+
+      dispatch(sendCommunityInvitation(community.id, {subject, message, emails, moderator}))
+      .then(({ error }) => {
+        if (error) return
+        trackEvent(INVITED_COMMUNITY_MEMBERS, {community})
+        dispatch(navigate(`/c/${community.slug}?invite-success`))
+      })
+    }
+
+    return <Modal title={`Invite Members to ${community.name}.`}
+      className='create-community-two'
+      standalone>
+      <div className='modal-input csv-upload'>
+        <label className='normal-label'>Import CSV File</label>
+        <label className='custom-file-upload'>
```

Comment:
We need some explanation here about what shape the CSV should be. If I'm understanding emailsFromCSVString correctly, it could be:

> The file should have a header row named "email" for the email column. Or it can be a file in which each line is a single email address.